acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -76,6 +76,16 @@ pub enum BlackBoxFuncCall {
         inputs: Vec<FunctionInput>,
         outputs: Vec<Witness>,
     },
+    VerifyProof {
+        key: Vec<FunctionInput>,
+        proof: Vec<FunctionInput>,
+        public_inputs: Vec<FunctionInput>,
+        key_hash: FunctionInput,
```

Comment:
Add comment specifying whether this is the proving/verification key hash and also why it is needed

Reply:
Commented